flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergSink.java

```diff
@@ -0,0 +1,774 @@
+          LOG.info("Distribute rows by equality fields, because there are equality fields set");
+          return input.keyBy(new EqualityFieldKeySelector(schema, flinkRowType, equalityFieldIds));
+        }
+
+      case HASH:
+        if (equalityFieldIds.isEmpty()) {
+          if (builder.table.spec().isUnpartitioned()) {
+            LOG.warn(
+                "Fallback to use 'none' distribution mode, because there are no equality fields set "
+                    + "and table is unpartitioned");
+            return input;
+          } else {
+            if (BucketPartitionerUtil.hasOneBucketField(spec)) {
+              return input.partitionCustom(
+                  new BucketPartitioner(spec),
+                  new BucketPartitionKeySelector(spec, schema, flinkRowType));
+            } else {
+              return input.keyBy(new PartitionKeySelector(spec, schema, flinkRowType));
+            }
+          }
+        } else {
+          if (spec.isUnpartitioned()) {
+            LOG.info(
+                "Distribute rows by equality fields, because there are equality fields set "
+                    + "and table is unpartitioned");
+            return input.keyBy(
+                new EqualityFieldKeySelector(schema, flinkRowType, equalityFieldIds));
+          } else {
+            for (PartitionField partitionField : spec.fields()) {
+              Preconditions.checkState(
+                  equalityFieldIds.contains(partitionField.sourceId()),
+                  "In 'hash' distribution mode with equality fields set, partition field '%s' "
+                      + "should be included in equality fields: '%s'",
+                  partitionField,
+                  builder.equalityFieldColumns);
+            }
+            return input.keyBy(new PartitionKeySelector(spec, schema, flinkRowType));
+          }
+        }
+
+      case RANGE:
+        if (equalityFieldIds.isEmpty()) {
+          LOG.warn(
+              "Fallback to use 'none' distribution mode, because there are no equality fields set "
+                  + "and {}=range is not supported yet in flink",
+              WRITE_DISTRIBUTION_MODE);
+          return input;
+        } else {
+          LOG.info(
+              "Distribute rows by equality fields, because there are equality fields set "
+                  + "and{}=range is not supported yet in flink",
+              WRITE_DISTRIBUTION_MODE);
+          return input.keyBy(new EqualityFieldKeySelector(schema, flinkRowType, equalityFieldIds));
+        }
+
+      default:
+        throw new RuntimeException("Unrecognized " + WRITE_DISTRIBUTION_MODE + ": " + mode);
+    }
+  }
+
+  /**
+   * Sets the currently used committer. Should be used for testing purposes only. Needs to be public
+   * as {@link IcebergSink} is used to test the {@link SinkCommitter} functionality.
+   *
+   * @param newCommitter The committer used for testing
+   */
+  public void setCommitter(Committer<SinkCommittable> newCommitter) {
```

Comment:
Not too elegant - could we find some workaround for setting the committer for tests, or inheriting from IcebergSink for testing, or something else?

Reply:
I made it package private and added `@VisibleForTesting`